src/Microsoft.Diagnostics.Monitoring.WebApi/Models/CallStackResults.cs

```diff
@@ -1,20 +1,46 @@
 ﻿// Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using Microsoft.Diagnostics.Monitoring.WebApi.Stacks;
 using System;
 using System.Collections.Generic;
+using System.Text;
 using System.Text.Json.Serialization;
 
 namespace Microsoft.Diagnostics.Monitoring.WebApi.Models
 {
     public class CallStackFrame
     {
         [JsonPropertyName("methodName")]
-        public string MethodName { get; set; } = string.Empty;
+        public string MethodNameWithGenericArgTypes
+        {
+            get
+            {
+                StringBuilder builder = new(MethodName);
+                NameFormatter.BuildGenericArgTypes(builder, FullGenericArgTypes);
+                return builder.ToString();
+            }
+            // Only intended for use test code.
+            set
```

Comment:
internal set

Reply:
When it's `internal`, `JsonSerializer` is unable to set this field in our tests when we deserialize our api output